### ISSUE TITLE
MuseSampler cannot handle large scores (again)

### ISSUE THREAD

**Issue by metasekk** [musescore/MuseScore#20450]
### Issue type

General playback bug

### Bug description

MuseSampler was recently updated from version 0.4.0.38 to 0.5.0.51. 

It re-introduced issue #15768 that was experienced with MuseSampler 0.3.2. When scores become large enough (~35 mn), the staves that use MuseSounds become permanently muted. Besides, depending on the content of the score before the cut-off, there can be drastic playback glitches, in which case some staves sound indefinitely, suggesting that the playback of the notes before the interruption is stacking up and never ends. 

### Steps to reproduce

1. Create a long score (duration : >~35 mn) and make sure you are using MuseSounds on several staves.
2. Write some notes on those staves.
3. Hear them be played until a certain length/bar is reached, after which the following notes won't sound.

### Screenshots/Screen recordings

I provide a [video](https://drive.google.com/file/d/1RVS-oM58RFYlDhMa_7Mx1wO_BRh6FAY4/view?usp=sharing) showing the steps described above and [the corresponding score](https://github.com/musescore/MuseScore/files/13661809/Score_issue_MuseSampler_05051.zip), as well as [another video](https://github.com/musescore/MuseScore/assets/120221700/86e459f8-8cfe-4038-b36d-e8ff932053ac) showing what it sounds like on a more realistic score.

### MuseScore Version

MuseSampler : v 0.5.0.51. Tested on Beta 4.2.0, Master 4.3.0 (Nightly-233470304-master-453c4c0-x86_64)

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows 10 Version 22H2, 64 bits, x64-based processor

### Additional context

Also see #15768 and #15094.

**Comment by bkunda** [comment]
Thanks for this @metasekk. The video doesn't seem to be there in the issue description. Can you please try adding it again?

**Comment by metasekk** [comment]
> Thanks for this @metasekk. The video doesn't seem to be there in the issue description. Can you please try adding it again?

Is it working now ?

**Comment by konradglas** [comment]
@metasekk Could you share that simplified reproduction score that you are creating in the first video? Thanks!

**Comment by shoogle** [comment]
I **wasn't** able to reproduce this with a 5 hour score with 9k measures: [Looong_score.mscz.zip](https://github.com/musescore/MuseScore/files/13662925/Looong_score.mscz.zip)

My Muse Sampler version is 0.5.0 as reported by MuseScore > Diagnostic > Muse Sampler.

@metasekk, do you experience the problem when you use my file? (You need to rename it to remove `.zip` from the end.)

Shortcut to jump to the end of the score is `Ctrl+End` (or `Cmd+Fn+Right` on macOS).

**Comment by metasekk** [comment]
I tested the file you provided in both 4.2.0 (latest nightly) and 4.3.0 (latest master nightly), and I still experience the issue (I can't hear the last notes, see this [video](https://github.com/musescore/MuseScore/assets/120221700/0d85b411-2db9-46da-a255-9d44a73de182)).

Interestingly enough, it doesn't happen in 4.1.1 on my side. However, all three versions (4.1.1, 4.2.0, 4.3.0) indicate that MuseSampler runs as "0.5.0".

**Comment by shoogle** [comment]
Interesting! Thanks for your analysis. I retested and found the same: it works in 4.1 but not in master or 4.2.

So it appears to be a MuseScore problem rather than a Muse Sampler problem. Maybe there's a 32-bit number involved somewhere, and it overflows for large scores?

**Comment by metasekk** [comment]
Could it have been caused by recent modifications in [libhandler.h](https://github.com/musescore/MuseScore/pull/19921/commits/bf9db818a022b6a4408dea6dbacf80500bf5e9d0#diff-36141142cb5aca5cc7c95c40d269a4cd6c8909ed7a68ace64dcb164568d16bd6) ? I remember that it contains conditions about the versions of MuseSampler.

**Comment by shoogle** [comment]
I just downgraded my Muse Sampler to 0.4.3 and I can no longer reproduce the problem, even in MuseScore master and 4.2.

So it seems to be the combination of **Muse Sampler 0.5.0** and **MuseScore master or 4.2** that is the problem.

To downgrade Muse Sampler, I had to:

1. Muse Hub > Settings > Quit.
2. Uninstall Muse Hub (beta version).
3. Install Muse Hub (non-beta version). *At this point you will still have the beta version of Muse Sampler.*
4. Go to `C:\Windows\System32` (macOS: `/usr/local/lib`, same for Linux too?)
5. Delete or rename the `MuseSamplerCoreLib.dll` file in that folder (macOS: `.dylib`, Linux: `.so`).
6. Re-open Muse Hub (it silently downloads and reinstalls Muse Sampler).
7. Relaunch MuseScore.
